File: app/src/main/java/com/example/user/fragmentbacktask/activity/BaseActivity.java
package com.example.user.fragmentbacktask.activity;

import android.content.Context;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;

import butterknife.ButterKnife;
import butterknife.Unbinder;

public abstract class BaseActivity extends AppCompatActivity {

    private Unbinder unbinder;
    public Context mContext;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(provideLayoutId());
        unbinder = ButterKnife.bind(this);
        mContext = this;
        initCreate(savedInstanceState);
    }

    public abstract int provideLayoutId();

    public abstract void initCreate(Bundle savedInstanceState);

    @Override
    protected void onDestroy() {
        super.onDestroy();
        unbinder.unbind();
    }
}

File: app/src/main/java/com/example/user/fragmentbacktask/kotlin/base/BaseActivity.kt
package com.example.user.fragmentbacktask.kotlin.base

import android.content.Context
import android.os.Bundle
import androidx.annotation.LayoutRes
import androidx.appcompat.app.AppCompatActivity
import butterknife.ButterKnife
import butterknife.Unbinder
import com.example.user.fragmentbacktask.TestApplication

abstract class BaseActivity : AppCompatActivity() {

    lateinit var mContext: Context
    var unbinder: Unbinder? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val id = provideLayoutId()
        if (id <= 0) {
            return
        }
        setContentView(id)
        unbinder = ButterKnife.bind(this)
        mContext = this
        initCreate(savedInstanceState)
    }

    @LayoutRes
    abstract fun provideLayoutId(): Int

    abstract fun initCreate(savedInstanceState: Bundle?)

    override fun onDestroy() {
        super.onDestroy()
        unbinder!!.unbind()
        TestApplication.getRefWatcher(this).watch(this)
    }

}
